dvc/remote/s3.py

```diff
@@ -234,3 +249,14 @@ def _generate_download_url(self, path_info, expires=3600):
         return self.s3.generate_presigned_url(
             ClientMethod="get_object", Params=params, ExpiresIn=int(expires)
         )
+
+    def _collect_dir(self, path_info):
+        root = path_info.path
+
+        return [
+            {
+                self.PARAM_CHECKSUM: entry["ETag"],
+                self.PARAM_RELPATH: os.path.relpath(entry["Key"], start=root),
```

Comment:
```suggestion
                self.PARAM_RELPATH: posixpath.relpath(entry["Key"], start=root),
```

Reply:
Will address this on another PR, @efiop, created an issue for it:  #2648

Reply:
@mroutis You are still using os.path to join things, which is incorrect and broke in tests on windows.